### DISCUSSION TITLE
Multiple "CodeStatements" for "CodeDom.CodeIterationStatement.IncrementStatement" possible?

### DISCUSSION THREAD

**Discussion by Gibson85** [dotnet/roslyn#68173] [Q&A]
Hi, this is not a proposal but a question. Imaging the following code:

`using System;
public class C {
    public void M() {
        int i = 0;
        int j = 0;
        for (i = 0, j = 0; (i < 10) && (j < 10); ++i, ++j)
        {
            Console.WriteLine(i.ToString());
        }
    }
}`

Now if I want to represent this "for" loop with the classes provided by the "System.CodeDom" namespace one should use the "CodeIterationStatement" class. But the "IncrementStatement" property of this class can hold only one statement. But in this case I am having to statements "++i" and "++j". How can this be solved? So long I could not figure out how to solve this. Any ideas how this could be done?

**Comment by CyrusNajmabadi** [comment]
FYI, CodeDom has not been updated in roughly 18 years for C#.  I would highly recommend not using it, and instead using the APIs in Roslyn instead (which can fully support this scenario).
